src/util/c_types_util.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018 Author: Malte Mues <mail.mues@gmail.com>
+
+/// \file
+/// This file contains functions, that should support test for underlying
+/// c types, in cases where this is required for anlysis purpose.
+#ifndef CPROVER_UTIL_C_TYPES_UTIL_H
+#define CPROVER_UTIL_C_TYPES_UTIL_H
+
+#include "arith_tools.h"
+#include "invariant.h"
+#include "std_types.h"
+#include "type.h"
+
+#include <algorithm>
+#include <string>
+
+/// This function checks, whether this has been a char type in the c program.
+inline bool is_c_char(const typet &type)
```

Comment:
I wonder if the naming of these new functions should be `is_blah_type` ? The thing being check is a *type*, not a value.